Agentic: Add ShippingValidator with address validation (5631)

danieldudzic opened the pull request:
### Description

Implements `ShippingValidator` covering three shipping validation scenarios:

  **Scenario 1: Invalid Shipping Address**
  - Validates required fields (street address, city, postal code)
  - Uses WooCommerce's native `WC_Validation::is_postcode()` for format validation

  **Scenario 2: PO Box Restriction**
  - Detects PO Box addresses
  - Validates signature-required items cannot ship to PO Boxes
  - Includes extensibility via `woocommerce_paypal_payments_agentic_commerce_item_requires_signature` filter

  **Scenario 3: Region Restricted**
  - Validates shipping country against WooCommerce allowed countries configuration
  - Provides localized country names in error messages

danieldudzic committed:
✨ Add context and resolution_options to ValidationIssue

danieldudzic committed:
✨ Add ShippingValidator for comprehensive shipping validation

danieldudzic committed:
🔧 Register ShippingValidator in dependency injection container

danieldudzic committed:
🔧 Add ShippingValidator to cart validation pipeline

danieldudzic committed:
✅ Add comprehensive tests for ShippingValidator

danieldudzic committed:
🐛 Fix test isolation issue with WC_Countries mocking

danieldudzic committed:
🔧 Add return type annotations to arrow functions for Psalm

stracker-phil commented on modules/ppcp-agentic-commerce/src/CartValidation/ShippingValidator.php:
```diff
@@ -0,0 +1,332 @@
+	 * @return string The country name, or the country code if name not found.
+	 */
+	private function get_country_name( string $country_code ): string {
+		if ( ! function_exists( 'WC' ) ) {
+			return $country_code;
+		}
+
+		$wc = WC();
+		if ( ! $wc || ! $wc->countries ) {
+			return $country_code;
+		}
```
To reduce the validation noise I suggest to add the `WooCommerce` instance as dependency, using the new service `woocommerce.core`

stracker-phil committed:
🔀 Merge branch ‘PCP-4891’

# Conflicts:
#	modules/ppcp-agentic-commerce/services.php
#	modules/ppcp-agentic-commerce/src/AgenticCommerceModule.php
